source/extensions/filters/http/cache/http_cache_utils.cc

```diff
@@ -182,6 +182,33 @@ SystemTime HttpCacheUtils::httpTime(const Http::HeaderEntry* header_entry) {
   }
   return {};
 }
+
+absl::optional<uint64_t> HttpCacheUtils::readAndRemoveLeadingDigits(absl::string_view& str) {
+  const char* ptr = str.data();
+  const char* limit = ptr + str.size();
+  uint64_t val = 0;
+
+  while (ptr < limit) {
```

Comment:
For clarity (and less worries about the dangers of C-style pointer arithmetic) writing this as a range iter over the str, tracking how many bytes were consumed:
```
uint32_t bytes_consumed = 0;
for (const char cur : str) {
  ++bytes_consumed;
  ...
```
I think most of the logic can stay as is, finishing with `str.remove_prefix(bytes_consumed);`

Note that I don't see a problem with your code; but it's just going to have to be scrutinized by a lot of people who need to check c-style pointer manipulation for potential security issues, and I think the pattern I give above is essentially the same; just easier to trust.